### DISCUSSION TITLE
Problem using resources file

### DISCUSSION THREAD

**Discussion by q-0q** [dotnet-state-machine/stateless#591] [Q&A] [answered]
> MissingManifestResourceException: Could not find any resources appropriate for the specified culture or the neutral culture.  Make sure "Stateless.StateConfigurationResources.resources" was correctly embedded or linked into assembly "Quantum.Simulation" at compile time, or that all the satellite assemblies required are loadable and fully signed.

Hello, I am having this issue when attempting to use `stateless` in my project. I am not familiar with resource files so I am not sure how to debug this.

I have placed the `stateless` source code into my directory, which is a Unity project. The Solution builds and runs fine while accessing the `stateless` library. The above exception is thrown only when trying to call `Fire` on my state machine.

I have tried setting the Build Action of `StateConfigurationResources.Designer.cs` to EmbeddedResource, but it reverts to the default (Compile) when building. I believe this may have to do with Unity regenerating the Solution, but I'm not certain. I am using JetBrains Rider.

**Comment by mclift** [comment]
There's a StackOverflow question that seems to describe a similar issue, [NuGet packages in Unity](https://stackoverflow.com/questions/53447595/nuget-packages-in-unity). The first two answers, using _NuGet for Unity_ and directly importing the .dll into a `plugin` folder, seem like they could apply here. Have you been down this path already?

If you're investigating other ways of building the project, only the .resx files should have their build action set to "EmbeddedResource". This build action embeds the resource files into Stateless.dll, so if you can find a way to import Stateless.dll into your Unity project then it should work.

Please let us know how it goes. I'm curious to find out if there's anything that would improve compatibility with Unity.

**Reply by q-0q** [reply, marked as answer]
Thank you so much for your response. I managed to get it working today with [this NuGet Unity package manager](https://github.com/GlitchEnzo/NuGetForUnity). I was off-track in trying to incorporate the raw source files into my project, the `.dll` was the way to go.

On the off-chance anyone in the future needs to do what I am doing and add the NuGet package to a preexisting Assembly, in addition to importing the package thru the NuGet manager, I also had to manually include the `.dll` in the `precompiledReferences` of my the Assembly's `.asmdef` file, like so:

```
{
    "name": "Quantum.Simulation",
     ...
    "precompiledReferences": [
        ...
        "Stateless.dll"
    ],
    ...
}
```
Cheers.

**Reply by mclift** [reply]
Thanks for letting us know the outcome, @q-0q! It's good to hear you got it working.
